src/main/java/fish/payara/micro/actions/MigrateToJakartaEE10Action.java

```diff
@@ -0,0 +1,116 @@
+
+import com.intellij.ide.impl.ProjectUtil;
+import com.intellij.openapi.actionSystem.AnActionEvent;
+import com.intellij.openapi.actionSystem.CommonDataKeys;
+import com.intellij.openapi.actionSystem.PlatformDataKeys;
+import com.intellij.openapi.editor.Editor;
+import com.intellij.openapi.fileChooser.FileChooser;
+import com.intellij.openapi.fileChooser.FileChooserDescriptor;
+import com.intellij.openapi.fileEditor.FileEditorManager;
+import com.intellij.openapi.fileEditor.OpenFileDescriptor;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.ui.Messages;
+import com.intellij.openapi.vfs.LocalFileSystem;
+import com.intellij.openapi.vfs.VfsUtil;
+import com.intellij.openapi.vfs.VirtualFile;
+import com.intellij.psi.PsiFile;
+import com.intellij.psi.search.FilenameIndex;
+import com.intellij.psi.search.GlobalSearchScope;
+import com.intellij.terminal.JBTerminalWidget;
+import fish.payara.micro.PayaraMicroProject;
+import fish.payara.micro.maven.MavenProject;
+import org.jetbrains.annotations.NotNull;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.logging.Logger;
+
+import static java.util.logging.Level.INFO;
+import static java.util.logging.Level.WARNING;
+
+public class MigrateToJakartaEE10Action extends MicroAction {
+    private static final Logger LOG = Logger.getLogger(MigrateToJakartaEE10Action.class.getName());
+
+    @Override
+    public void actionPerformed(@NotNull AnActionEvent actionEvent) {
+        VirtualFile srcFile = PlatformDataKeys.VIRTUAL_FILE.getData(actionEvent.getDataContext());
+        assert srcFile != null;
+        FileChooserDescriptor fileDescriptor = new FileChooserDescriptor(false,true,
+                false,false,false,false)
+                .withTitle(srcFile.isDirectory() ? "Choose the New Project Destination Folder" :
+                        "Choose the New File Destination Folder");
+        VirtualFile destFolder = FileChooser.chooseFile(fileDescriptor, actionEvent.getProject(), srcFile);
+        assert destFolder != null;
+        String destinationPath = getDestinationPath(destFolder, srcFile);
+
+        final Project project = CommonDataKeys.PROJECT.getData(actionEvent.getDataContext());
+        assert project != null;
+        PsiFile[] poms = FilenameIndex.getFilesByName(project, "pom.xml", GlobalSearchScope.projectScope(project));
+        PayaraMicroProject microProject = new MavenProject(project, poms[0]);
+        String projectName = project.getName();
+        JBTerminalWidget terminal = getTerminal(project, projectName);
+        if (terminal != null) {
+            Messages.showMessageDialog(
+                    "Generating: " + destinationPath,
+                    "Confirmation",
+                    Messages.getInformationIcon());
+            executeCommand(terminal, microProject.getTransformCommand(srcFile.getPath(), destinationPath));
```

Comment:
Also, add `executeCommand` to the worker thread and follow the guidelines from IntelliJ Platform SDK to use UI threads for writing the files.
https://plugins.jetbrains.com/docs/intellij/general-threading-rules.html#do-not-perform-long-operations-in-ui-thread

Reply:
now using an ExecutorService and submiting 2 threads